src/mon/OSDMonitor.cc

```diff
@@ -3524,16 +3524,11 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
     if (osdmap.test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
       int full, nearfull;
       osdmap.count_full_nearfull_osds(&full, &nearfull);
-      if (full > 0) {
-	ostringstream ss;
-	ss << full << " full osds(s)";
-	summary.push_back(make_pair(HEALTH_ERR, ss.str()));
-      }
-      if (nearfull > 0) {
-	ostringstream ss;
-	ss << nearfull << " nearfull osds(s)";
-	summary.push_back(make_pair(HEALTH_WARN, ss.str()));
-      }
+      ostringstream ss;
+
+      if (full > 0) ss << full << " full osds(s)";
+      if (nearfull > 0) ss << nearfull << " nearfull osds(s)";
+      summary.push_back(make_pair(((full > 0) ? HEALTH_ERR : HEALTH_WARN), ss.str()));
```

Comment:
This doesnt' look right; summary should only be updated if full > 0 or nearfull > 0